### PR TITLE
Delete the archival workflow code

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
This PR deletes all the code for the old workflow-based workflow archival feature. 

**Why?**
Originally, the tool to archive workflows was itself written as a workflow. There were some complications that arose from this, so we decided to build the archival system closer to the metal as its own queue similar to the timer and transfer queues. In a previous release, we disabled the archival workflow, and enabled the new archival queue. Now that this process is done, we can delete all the dead code for the archival workflow.

**How did you test it?**
I just searched for "archiv" in our codebase and looked at every occurrence. 

**Potential risks**
Maybe there are people still using the old archival workflow, and they'll now need to migrate.

**Is hotfix candidate?**
I don't think so.

### CODE REVIEW

File: service/history/workflow/task_generator.go [hunk 1]
```diff
@@ -193,7 +190,7 @@ func (r *TaskGeneratorImpl) GenerateWorkflowCloseTasks(
 				Version:     currentVersion,
 			},
 		)
-		if r.archivalQueueEnabled() {
+		if r.archivalEnabled() {
```
Comment (yycptt) [inline service/history/workflow/task_generator.go:193]: so if we are deleting an open workflow, that workflow won't be archived right? just want to confirm my understanding.

File: common/persistence/serialization/task_serializer.go [hunk 2]
```diff
@@ -552,7 +552,9 @@ func (s *TaskSerializer) transferCloseTaskToProto(
 		DeleteAfterClose:        closeTask.DeleteAfterClose,
 		TaskDetails: &persistencespb.TransferTaskInfo_CloseExecutionTaskDetails_{
 			CloseExecutionTaskDetails: &persistencespb.TransferTaskInfo_CloseExecutionTaskDetails{
-				CanSkipVisibilityArchival: closeTask.CanSkipVisibilityArchival,
+				// We set this to true even though it's no longer checked in case someone downgrades to a version that
+				// still checks this field.
+				CanSkipVisibilityArchival: true,
```
Comment (yycptt) [inline common/persistence/serialization/task_serializer.go:557]: Can we remove those flags in later releases?

**Review by yycptt — APPROVED** [review]

**Comment by yycptt** [comment]
Actually I think we need some release notes for this change. 

If the old workflow archival logic is still in use, user needs to make sure all of those archival workflows complete before upgrading. Probably by upgrading to 1.20 or 1.21 or 1.22, enable durable archival feature flag, and wait for all archival workflow to complete. We need to provide a visibility query for verifying this.

File: common/persistence/serialization/task_serializer.go (hunk 2 above)
Comment (MichaelSnowden) [reply common/persistence/serialization/task_serializer.go:557]: Yes.

File: service/history/workflow/task_generator.go (hunk 1 above)
Comment (MichaelSnowden) [reply service/history/workflow/task_generator.go:190]: Yeah the manual delete skips archival.

**Pull request merged**